src/main/java/pl/sda/javapoz/service/impl/ProductOrderServiceImpl.java

```diff
@@ -51,43 +57,45 @@ public ProductOrderEntity getPriceOfOrderedProduct(ProductOrderEntity productOrd
     @Override
     public Double getPriceOfOrderedProducts(List<ProductOrderEntity> productOrders) {
         Double[] sum = {0.0};
-        productOrders.forEach(product -> sum[0] += product.getCombinedPrice());
+        // productOrders.forEach(product -> sum[0] += product.getCombinedPrice());
 
         return sum[0];
     }
 
-
     @Override
-    public boolean isProductAvailableToOrder(Long id, String dateFilter) {
-        List<ProductOrderEntity> orders = findProductOrdersByProductId(id);
-        boolean availableToOrder = true;
+    public Integer countOrdersProductInPeriod(Long productId, Date productOrderStart, Date productOrderEnd) {
+        List<ProductOrderEntity> orders = findProductOrdersByProductId(productId);
+
+        int count = 0;
         for (ProductOrderEntity order : orders) {
             Date orderStart = order.getOrderStart();
             Date orderEnd = order.getOrderEnd();
-            String[] dates = DateFilter.filterData(dateFilter);
-            Date productOrderStart = new Date(dates[0]);
-            Date productOrderEnd = new Date(dates[1]);
-
             if (productOrderStart.before(orderEnd) && productOrderEnd.after(orderStart)) {
-                availableToOrder = false;
-                break;
+                count += order.getProducts().stream()
+                        .filter(productEntity -> productEntity.getId().equals(productId))
+                        .collect(Collectors.toList())
+                        .size();
             }
         }
-        return availableToOrder;
+
+        return count;
     }
 
     @Override
-    public boolean isProductAvailableToOrder(Long id, Date productOrderStart, Date productOrderEnd) {
-        List<ProductOrderEntity> orders = findProductOrdersByProductId(id);
-        boolean availableToOrder = true;
-        for (ProductOrderEntity order : orders) {
-            Date orderStart = order.getOrderStart();
-            Date orderEnd = order.getOrderEnd();
-            if (productOrderStart.before(orderEnd) && productOrderEnd.after(orderStart)) {
-                availableToOrder = false;
-            }
-        }
-        return availableToOrder;
```

Comment:
to mozna umiescic w jednej metodzie, wydaje mi sie ze nie ma potrzeby ktora uzasadnialaby takie rozdzielenie